sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -126,7 +126,14 @@ private[sql] case class AggregateExpression(
     AttributeSet(childReferences)
   }
 
-  override def toString: String = s"($aggregateFunction,mode=$mode,isDistinct=$isDistinct)"
+  override def toString: String = {
+    val agg = aggregateFunction
+    if (isDistinct) {
+      s"${mode}_${agg.prettyName}(distinct ${agg.argString(false)})"
+    } else {
+      s"${mode}_${agg.simpleString}"
+    }
+  }
```

Comment:
Personally I'd prefer `:` instead of `_` as separator between `mode` and `agg`. Because `_` is a valid identifier character, while `:` is not, thus it's more obvious that `partial:` is a prefix rather part of the aggregate function name.

(But that's pretty subjective.)